Source/JavaScriptCore/runtime/IteratorOperations.h

```diff
@@ -63,6 +67,43 @@ JS_EXPORT_PRIVATE IterationMode getIterationMode(VM&, JSGlobalObject*, JSValue i
 JS_EXPORT_PRIVATE IterationMode getIterationMode(VM&, JSGlobalObject*, JSValue iterable, JSValue symbolIterator);
 
 
+template<typename CallBackType>
+static ALWAYS_INLINE void forEachInMapStorage(VM& vm, JSGlobalObject* globalObject, JSCell* storageCell, JSMap::Helper::Entry startEntry, IterationKind iterationKind, NOESCAPE const CallBackType& callback)
+{
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* storage = jsCast<JSMap::Storage*>(storageCell);
+    JSMap::Helper::Entry entry = startEntry;
+
+    while (true) {
+        storageCell = JSMap::Helper::nextAndUpdateIterationEntry(vm, *storage, entry);
+        if (storageCell == vm.orderedHashTableSentinel())
+            break;
+
+        storage = jsCast<JSMap::Storage*>(storageCell);
+        entry = JSMap::Helper::iterationEntry(*storage) + 1;
+        JSValue entryKey = JSMap::Helper::getIterationEntryKey(*storage);
```

Comment:
maybe it should only get the specific value if it would be used? like if keys || entries, then entryKey, if values || entries, then entryValue?